### PR TITLE
binary search

### PR DESCRIPTION
**Pull request by AliaksandrZahorski**

### CODE REVIEW

File: binary_search/script.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    el = arr[mid];
+    if (key > el) {
+      lo = mid + 1;
+    } else if (key < el) {
+      up = mid - 1;
+    } else {
+      return mid;
+    }
+  }
+  return -mid - 1;
```
Comment (NikitaSedyx) [inline binary_search/script.js]: it should return undefined or -1 because element is not found

File: binary_search/script.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+var sortedNumbers = [1, 2, 3, 4];
+
+console.log(binarySearch(sortedNumbers, 1));
+console.log(binarySearch(sortedNumbers, 2));
+console.log(binarySearch(sortedNumbers, -5));
+console.log(binarySearch(sortedNumbers, 5));
+console.log(binarySearch(sortedNumbers, 2.5));
+
+function binarySearch(arr, key) {
+  var lo = 0, up = arr.length - 1, mid, el;
```
Comment (NikitaSedyx) [inline binary_search/script.js]: I'd suggest to change variable names, it difficult to understand what they mean at the beginning.
also it's better to break line when you initialize variables
```
var lo = 0,
      up = arr.length - 1,
      mid, el;
```
or
```
var lo = 0;
var up = arr.length - 1;
var mid; //var mid, el;
var el;
```
imo the #2 looks better

**Review by NikitaSedyx — APPROVED** [review]
function works

**Comment by NikitaSedyx** [comment]
could you try to implement the same function but with recursion?

File: binary_search/script.js (hunk 2 above)
Comment (AliaksandrZahorski) [reply binary_search/script.js]: fixed

File: binary_search/script.js (hunk 1 above)
Comment (AliaksandrZahorski) [reply binary_search/script.js]: This is more like "java style" of binary search function)

**Comment by AliaksandrZahorski** [comment]
Sure. Here is link for this version of binary search.

https://github.com/AliaksandrZahorski/AliaksandrZahorski.github.io/pull/4

2017-04-04 18:08 GMT+03:00 Mikita Siadykh <notifications@github.com>:

**Pull request merged**